smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -662,15 +671,18 @@ private void parseInlineableOperationMember(ShapeId id, String suffix, Consumer<
             expect('=');
             clearPendingDocs();
             ws();
-            consumer.accept(parseInlineStructure(id.getName() + suffix));
+            consumer.accept(parseInlineStructure(id.getName() + suffix, defaultTrait));
         } else {
             ws();
             modelFile.addForwardReference(ParserUtils.parseShapeId(this), consumer);
         }
     }
 
-    private ShapeId parseInlineStructure(String name) {
+    private ShapeId parseInlineStructure(String name, TraitEntry defaultTrait) {
         List<TraitEntry> traits = parseDocsAndTraits();
+        if (defaultTrait != null) {
+            traits.add(defaultTrait);
```

Comment:
What happens if an `@input`/`@output` is applied manually? I don't see a test for that either.

Reply:
The dupe is ignored. Added a test case for it.